src/koopman/koopman.jl

```diff
@@ -0,0 +1,200 @@
+mutable struct Koopman{T, FP, FB, FC} <: abstractKoopmanOperator
+
+    A::AbstractArray{T,2} # Approximation of the operator, Transition
+
+    ψ::FP # Map from states to observable
+    B::FB # Input mapping
+    C::FC # Output mapping
+
+    λ::AbstractArray{Complex{T},1} # Eigenvalues (discrete time)
+    ω::AbstractArray{Complex{T},1} # Frequencies
+    ϕ::AbstractArray{Complex{T},2} # Modes
+
+    # For online DMD
+    Qₖ::AbstractArray{T,2}
+    Pₖ::AbstractArray{T,2}
+
+    # Information
+    islifted::Bool
+    iscontrolled::Bool
+end
+
+
+# Purely linear system
+function Koopman(A::AbstractArray{T,2}; Q::AbstractArray{T,2} = Array{T}(undef, 0, 0), P::AbstractArray{T,2} = Array{T}(undef, 0, 0), dt::R = 0.0) where {T <: Real, R <: Real}
+    # Eigen Decomposition for solution
+    Λ, W = eigen(A)
+
+    if dt > 0.0
+        # Casting Complex enforces results
+        ω = log.(Complex.(Λ)) / dt
+    else
+        ω = eltype(A)[]
+    end
+
+    ψ(u, p, t) = identity(u)
+    C(u, p, t) = identity(u)
+    B(u, p, t) = zeros(u)
+
+    return Koopman(A, ψ, B, C, Complex.(Λ), Complex.(ω), Complex.(W), Q, P, false, false)
+end
+
+# Autonomos
+function Koopman(A::AbstractArray{T,2}, ψ, C ; B = nothing, Q::AbstractArray{T,2} = Array{T}(undef, 0, 0), P::AbstractArray{T,2} = Array{T}(undef, 0, 0), dt::R = 0.0) where {T <: Real, R <: Real}
+    # Eigen Decomposition for solution
+    Λ, W = eigen(A)
+
+    if dt > 0.0
+        # Casting Complex enforces results
+        ω = log.(Complex.(Λ)) / dt
+    else
+        ω = eltype(A)[]
+    end
+
+    b(u,p,t) = isnothing(B) ? zeros(u) : B(u,p,t)
+
+    return Koopman(A, ψ, b, C ,Complex.(Λ), Complex.(ω), Complex.(W), Q, P, true, !isnothing(B))
+end
+
+function Koopman(A::AbstractArray{T,2}, ψ::AbstractArray{T,2}, C::AbstractArray{T,2} ; B = nothing, Q::AbstractArray{T,2} = Array{T}(undef, 0, 0), P::AbstractArray{T,2} = Array{T}(undef, 0, 0), dt::R = 0.0) where {T <: Real, R <: Real}
+    psi(u,p,t) = ψ*u
+    c(u,p,t) = C*u
```

Comment:
probably good to have alternative dispatches as well, `c(du,u,p,t) = mul!(du,C,u)` and the other.